Runtime/AvatarCreator/Scripts/Extensions/TaskExtensions.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Threading.Tasks;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    public static class TaskExtensions
+    {
+        public const string ON_REQUEST_CANCELLED_MESSAGE = "Request was cancelled"; 
+        public static async Task HandleCancellation(Task taskToWait, Action onSuccess = null)
+        {
+            try
+            {
+                await taskToWait;
+                onSuccess?.Invoke();
+            }
+            catch (Exception ex)
```

Comment:
If you want to listen for cancellations you can do this
```suggestion
            catch (OperationCanceledException ex)
```
Then you don't need to check the message name and it will catch all cancellations

Reply:
If the intention is to only catch cancellation token exceptions, and not catch anything else, then catching that specific error is generally the best way to go.

https://learn.microsoft.com/en-us/dotnet/standard/exceptions/how-to-use-specific-exceptions-in-a-catch-block

Reply:
Ah, but I see, you're saying this is catching a middle layer of response exceptions that we've defined, so the actual type of the exception is lost before it reaches here?

Reply:
yep